external-plugins/rehearse/plugin/handler/handler.go

```diff
@@ -0,0 +1,504 @@
+		return false
+	}
+	return isMember || github.HasLabel("ok-to-test", pr.Labels)
+}
+
+func (h *gitHubEventsHandler) handlePullRequestUpdateEvent(log *logrus.Entry, event *github.PullRequestEvent) {
+	defer func() {
+		if r := recover(); r != nil {
+			h.logger.Warnf("Recovered during handling of a pull request event: %s", event.GUID)
+		}
+	}()
+
+	if !h.alwaysRun {
+		return
+	}
+
+	log.Infof("Handling updated pull request: %s [%d]", event.Repo.FullName, event.PullRequest.Number)
+
+	if !h.shouldActOnPREvent(event) {
+		log.Infoln("Skipping event. Not of our interest.")
+		return
+	}
+
+	if !h.validateEventUser(event.Repo.FullName, event.Sender.Name, event.PullRequest.Number) {
+		log.Infoln("Skipping event. User is not authorized.")
+		return
+	}
+
+	h.handleRehearsalForPR(log, &event.PullRequest, event.GUID)
+}
+
+func (h *gitHubEventsHandler) handleRehearsalForPR(log *logrus.Entry, pr *github.PullRequest, eventGUID string) {
+	repo, org, err := gitv2.OrgRepo(pr.Head.Repo.FullName)
+	if err != nil {
+		log.WithError(err).Errorf("Could not parse repo name: %s", pr.Head.Repo.FullName)
+		return
+	}
+	log.Infoln("Generating git client")
+	git, err := h.gitClientFactory.ClientFor(repo, org)
+	if err != nil {
+		return
+	}
+
+	log.Infoln("Fetching PR base ref", pr.Base.SHA)
+	err = git.FetchRef(pr.Base.SHA)
+	if err != nil {
+		log.WithError(err).Error("Could not fetch pull request's base ref.")
+		return
+	}
+	log.Infoln("Fetching PR head ref", pr.Head.SHA)
+	err = git.FetchRef(pr.Head.SHA)
+	if err != nil {
+		log.WithError(err).Error("Could not fetch pull request's head ref.")
+		return
+	}
+	log.Infoln("Getting diff")
+	changedFiles, err := git.Diff(pr.Head.SHA, pr.Base.SHA)
+	if err != nil {
+		log.WithError(err).Error("Could not calculate diff for PR.")
+		return
+	}
+	log.Infoln("Changed files:", changedFiles)
+	changedJobConfigs, err := h.modifiedJobConfigs(changedFiles)
+	if err != nil {
+		log.WithError(err).Error("Could not calculate absolute paths for modified job configs")
+		return
+	}
+	log.Infoln("Changed job configs:", changedJobConfigs)
+	headConfigs, err := h.loadConfigsAtRef(changedJobConfigs, git, pr.Head.SHA)
+	if err != nil {
+		log.WithError(err).Errorf(
+			"Could not load job configs from head ref: %s", pr.Head.SHA)
+	}
+
+	baseConfigs, err := h.loadConfigsAtRef(changedJobConfigs, git, pr.Base.SHA)
+	if err != nil {
+		log.WithError(err).Errorf(
+			"Could not load job configs from base ref: %s", pr.Base.SHA)
+	}
+	log.Infoln("Base configs:", baseConfigs)
+
+	prowjobs := h.generateProwJobs(headConfigs, baseConfigs, pr, eventGUID)
+	log.Infof("Will create %d jobs", len(prowjobs))
+	for _, job := range prowjobs {
+		if job.Labels == nil {
+			job.Labels = make(map[string]string)
+		}
+		job.Labels["rehearsal"] = "true"
+		job.Labels["rehearsal-for-pull-request"] = strconv.Itoa(pr.Number)
+		job.Spec.Job = strings.Join([]string{"rehearsal", job.Spec.Job}, "-")
```

Comment:
You also need to set `Spec.Context` to `strings.Join([]string{"rehearsal", job.Spec.Job}, "-")` to report it with a `rehearsal` prefix on the github PR. This way one can clearly see that this is a rehearsal job on the PR.